### PR TITLE
Add command sw:system:config:set

### PR DESCRIPTION
**Pull request by jkrzefski**
### 1. Why is this change necessary?
This PR adds a command to configure basic settings. The signature looks like this:

```
sw:system:config:set [--shopId[=SHOPID]] [--] <name> <value>
```

### 2. What does this change do, exactly?
In short: This command sets basic settings. You have to specify a configuration key and its new value. Optionally, you can also specify a shop-id. If you don't specify one, the default shop-id is assumed.

Internally the command creates an HttpKernel and dispatches a request to the `backend/config` controller. So this will call the exact same business logic as the backend would, including all the events that are dispatches during this controller action. This way, a maximum of compatibility is ensured.

### 3. Describe each step to reproduce the issue or behaviour.
```
bin/console sw:system:config:set shopName 'It works!'
```

### 4. Please link to the relevant issues (if any).
None.

### 5. Which documentation changes (if any) need to be made because of this PR?
None.

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by mitelg** [comment]
hey @jkrzefski 

thanks for your contribution :+1: 

We really appreciate your new feature, but we cannot accept how it was implemented. As you already might know this is very hacky :wink: 

Instead we would like to extract the need logic from the controller into a new service. This should be quite easy doable, as the concerning method is private: https://github.com/shopware/shopware/blob/5.7/engine/Shopware/Controllers/Backend/Config.php#L1392

With this done, you can also use this new service in your command and there is no need to boot a new kernel and create a dummy request and so on...

File: engine/Shopware/Commands/SystemConfigSetCommand.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class SystemConfigSetCommand extends ShopwareCommand
+{
+    protected function configure(): void
+    {
+        $this
+            ->setName('sw:system:config:set')
```
Comment (mitelg) [inline engine/Shopware/Commands/SystemConfigSetCommand.php]:
```suggestion
            ->setName('sw:config:set')
```

sounds like the SW6 command :grimacing:

**Comment by jkrzefski** [comment]
@mitelg Thank you for your feedback 😄

Unfortunately, I don't see this task as that "easy". Extracting the logic from the controller into a dedicated service would normally be my go-to solution, but the way I see it, that would introduce a **breaking change**. The events `Shopware_Controllers_Backend_Config_Before_Save_Config_Element` and `Shopware_Controllers_Backend_Config_After_Save_Config_Element` both need an instance of the backend controller for their event arguments. (https://github.com/shopware/shopware/blob/5.7/engine/Shopware/Controllers/Backend/Config.php#L1450-L1466) A service could not possibly provide such an instance.

(I could probably make it work, but that would be even more "hacky" than this proposed solution.)

[I was told, we're not doing breaking changes in Shopware 5 anymore.](https://github.com/shopware/shopware/pull/2311#issuecomment-585827456) So what should we do? Is the only possible solution to not support the events when the command is used?

**Comment by mitelg** [comment]
as discussed on Slack, only implement a very thin layer in the command to set the value to the element

**Comment by jkrzefski** [comment]
@mitelg I changed the implementation as we discussed on Slack. There is one slight change I added: Instead of always using `json_decode` on the input value, this behavior can now be toggled with a new option `--decode`.

**Comment by mitelg** [comment]
thanks for your contribution @jkrzefski  :+1: :tada: :blue_heart:

don't be confused, we merged it on the next branch, which will be merged back to the main branch in a few days

**Pull request closed without merging**